src/language-js/needs-parens.js

```diff
@@ -1,5 +1,6 @@
 "use strict";
 
+/** @type {import("assert")} */
```

Comment:
Why is this needed? `assert` is Node's builtin module. Its types should come from`@types/node`. `require("assert");` should just work.

Reply:
I would get these errors without this type import, `yarn add --dev @types/node` does not seem to make any difference here:

```
src/language-js/needs-parens.js:302:13 - error TS2775: Assertions require every name in the call target to be declared with an explicit type annotation.

302             assert.strictEqual(parent.right, node);
                ~~~~~~~~~~~~~~~~~~

  src/language-js/needs-parens.js:3:7
    3 const assert = require("assert");
            ~~~~~~
    'assert' needs an explicit type annotation.
```

I guess this is something with the TypeScript compiler. Any ideas?

Reply:
This throws since this PR I sent https://github.com/prettier/prettier/pull/8702/files

Reply:
This is a bug in TS: https://github.com/microsoft/TypeScript/issues/38379

Reply:
`/** @type {import("assert")} */` can stay for now, but we need to use `@types/node` instead of `@types/assert`. I'm not sure why the latter even exists.

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/assert/index.d.ts - seems to be for the Browserify assert package